RFC-0007-plan-constraints-grammar.md

```diff
@@ -0,0 +1,168 @@
+# **RFC007 for Presto**
+
+## Grammar for Plan Constraints
```

Comment:
I would call this feature Plan Hints rather than Plan Constraints. The optimizer may not follow the hints (though we should issue a warning if we don't)

Reply:
>I would call this feature Plan Hints rather than Plan Constraints.  

Sure, will rename
Agreed on providing a warning if some/all of the plan hint is not applied